### ISSUE TITLE
scriptSort throws script lang not supported error (groovy)

### ISSUE THREAD

**Issue by MichaelViveros** [sksamuel/elastic4s#904]
I'm trying to sort the results of a query using a computed value (so I need to use scripting) but I keep getting an error when I call `scriptSort` about the groovy scripting language not being supported. I'm using elastic4s [v2.4.0](https://github.com/sksamuel/elastic4s/tree/v2.4.0).

Here's a simple example I made to illustrate the error. I have `bands` with names `coldplay` and `badbadnotgood` and I want to sort them by a computed value which is the length of their names (`coldplay` -> 8, `badbadnotgood` -> 13). I combined the sample code from the [README](https://github.com/sksamuel/elastic4s/tree/v2.4.0) and from [ScriptTest.scala](https://github.com/sksamuel/elastic4s/blob/v2.4.0/elastic4s-core-tests/src/test/scala/com/sksamuel/elastic4s/ScriptTest.scala) to get:

```scala
package essample

import com.sksamuel.elastic4s.ElasticClient
import com.sksamuel.elastic4s.ElasticDsl._
import org.elasticsearch.common.settings.Settings
import org.elasticsearch.search.sort.SortOrder

object EsSampleApp extends App {

  val client = {
    val esSettings =
      Settings.settingsBuilder().put("path.home", "~/IdeaProjects/ElasticsearchSample/.esdata")
    ElasticClient.local(esSettings.build)
  }

  client.execute { index into "bands" / "artists" fields "name"->"coldplay" }.await
  client.execute { index into "bands" / "artists" fields "name"->"badbadnotgood" }.await

  Thread.sleep(2000)

  val sorted = client.execute {
    search in "bands" / "artists" query matchAllQuery sort {
      scriptSort("""if (_source.containsKey('name')) _source['name'].length else 0""") typed "number" order SortOrder.DESC
    }
  }.await
  println(sorted)


}
```

My build.sbt:
```
name := "ElasticsearchSample"

version := "1.0"

scalaVersion := "2.11.8"

libraryDependencies += "com.sksamuel.elastic4s" %% "elastic4s-core" % "2.4.0"
```

It compiles fine but when I run it I get the following error message:
```error
Exception in thread "main" Failed to execute phase [query], all shards failed; shardFailures {[bSPqQgMVTFeCFrWE7Xol6g][bands][0]: RemoteTransportException[[Smasher][local[1]][indices:data/read/search[phase/query]]]; nested: SearchParseException[failed to parse search source [{"query":{"match_all":{}},"sort":[{"_script":{"script":{"inline":"if (_source.containsKey('name')) _source['name'].length else 0"},"type":"number","order":"desc"}}]}]]; nested: IllegalArgumentException[script_lang not supported [groovy]]; }{[bSPqQgMVTFeCFrWE7Xol6g][bands][1]: RemoteTransportException[[Smasher][local[1]][indices:data/read/search[phase/query]]]; nested: SearchParseException[failed to parse search source [{"query":{"match_all":{}},"sort":[{"_script":{"script":{"inline":"if (_source.containsKey('name')) _source['name'].length else 0"},"type":"number","order":"desc"}}]}]]; nested: IllegalArgumentException[script_lang not supported [groovy]]; }{[bSPqQgMVTFeCFrWE7Xol6g][bands][2]: RemoteTransportException[[Smasher][local[1]][indices:data/read/search[phase/query]]]; nested: SearchParseException[failed to parse search source [{"query":{"match_all":{}},"sort":[{"_script":{"script":{"inline":"if (_source.containsKey('name')) _source['name'].length else 0"},"type":"number","order":"desc"}}]}]]; nested: IllegalArgumentException[script_lang not supported [groovy]]; }{[bSPqQgMVTFeCFrWE7Xol6g][bands][3]: RemoteTransportException[[Smasher][local[1]][indices:data/read/search[phase/query]]]; nested: SearchParseException[failed to parse search source [{"query":{"match_all":{}},"sort":[{"_script":{"script":{"inline":"if (_source.containsKey('name')) _source['name'].length else 0"},"type":"number","order":"desc"}}]}]]; nested: IllegalArgumentException[script_lang not supported [groovy]]; }{[bSPqQgMVTFeCFrWE7Xol6g][bands][4]: RemoteTransportException[[Smasher][local[1]][indices:data/read/search[phase/query]]]; nested: SearchParseException[failed to parse search source [{"query":{"match_all":{}},"sort":[{"_script":{"script":{"inline":"if (_source.containsKey('name')) _source['name'].length else 0"},"type":"number","order":"desc"}}]}]]; nested: IllegalArgumentException[script_lang not supported [groovy]]; }
	at org.elasticsearch.action.search.AbstractSearchAsyncAction.onFirstPhaseResult(AbstractSearchAsyncAction.java:206)
	at org.elasticsearch.action.search.AbstractSearchAsyncAction$1.onFailure(AbstractSearchAsyncAction.java:152)
	at org.elasticsearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:46)
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.processException(TransportService.java:872)
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:850)
	at org.elasticsearch.transport.TransportService$4.onFailure(TransportService.java:387)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:39)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: [script_lang not supported [groovy]]; nested: IllegalArgumentException[script_lang not supported [groovy]];
	at org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:386)
	at org.elasticsearch.action.search.SearchPhaseExecutionException.guessRootCauses(SearchPhaseExecutionException.java:152)
	at org.elasticsearch.action.search.SearchPhaseExecutionException.getCause(SearchPhaseExecutionException.java:99)
	at java.lang.Throwable.printStackTrace(Throwable.java:665)
	at java.lang.Throwable.printStackTrace(Throwable.java:643)
	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1061)
	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1052)
	at java.lang.Thread.dispatchUncaughtException(Thread.java:1952)
Caused by: java.lang.IllegalArgumentException: script_lang not supported [groovy]
	at org.elasticsearch.script.ScriptService.getScriptEngineServiceForLang(ScriptService.java:212)
	at org.elasticsearch.script.ScriptService.compile(ScriptService.java:242)
	at org.elasticsearch.script.ScriptService.search(ScriptService.java:456)
	at org.elasticsearch.search.sort.ScriptSortParser.parse(ScriptSortParser.java:137)
	at org.elasticsearch.search.sort.SortParseElement.addCompoundSortField(SortParseElement.java:142)
	at org.elasticsearch.search.sort.SortParseElement.parse(SortParseElement.java:85)
	at org.elasticsearch.search.SearchService.parseSource(SearchService.java:856)
	at org.elasticsearch.search.SearchService.createContext(SearchService.java:667)
	at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:633)
	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:377)
	at org.elasticsearch.search.action.SearchServiceTransportAction$SearchQueryTransportHandler.messageReceived(SearchServiceTransportAction.java:368)
	at org.elasticsearch.search.action.SearchServiceTransportAction$SearchQueryTransportHandler.messageReceived(SearchServiceTransportAction.java:365)
	at org.elasticsearch.transport.TransportRequestHandler.messageReceived(TransportRequestHandler.java:33)
	at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:77)
	at org.elasticsearch.transport.TransportService$4.doRun(TransportService.java:376)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

I also tried adding a `groovy` dependency in my build.sbt that was in elastic4s' [build.sbt](https://github.com/sksamuel/elastic4s/blob/v2.4.0/build.sbt#L25) but it didn't work:
```
libraryDependencies += "org.elasticsearch.module" % "lang-groovy" % "2.4.1"
```

**Comment by sksamuel** [comment]
Does your actual elasticsearch instance have support for groovy scripting?

**Comment by MichaelViveros** [comment]
Looks like it didn't, sorry about that.

I'm setting up another project that uses elastic4s and after I set it up it's elasticsearch instance didn't have any module/plugin for groovy scripting. So I downloaded elasticsearch from the elasticsearch website which had modules for groovy and other scripting langauges and then I pointed the project's elasticsearch instance to the downloaded one and it works fine.

Thanks for your help.
